src/v0/destinations/amazon_audience/transform.js

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+const { InstrumentationError } = require('@rudderstack/integrations-lib');
+const { handleRtTfSingleEventError } = require('../../util');
+const { batchEvents, buildResponseWithUsers, getUserDetails } = require('./utils');
+/**
+ * This function returns the user traits list required in request for
+ * making a call to create a group of users in amazon_audience
+ * @param {*} record
+ * @param {*} destination
+ * @param {*} metadata
+ */
+const processRecord = (record, config) => {
+  const { fields, action, type } = record;
+  if (type !== 'record') {
+    throw new InstrumentationError(`[AMAZON AUDIENCE]: ${type} is not supported`);
+  }
+  return { user: getUserDetails(fields, config), action: action !== 'delete' ? 'add' : 'remove' };
```

Comment:
In case of update action -  we should remove first and then add. Otherwise it will result in duplicates right?